### PR TITLE
Add descriptive error message for implicit conversion conflict

### PR DESCRIPTION
**Pull request by aaabramov**
Fixes https://github.com/RustedBones/akka-http-metrics/issues/117

### CODE REVIEW

**Comment by aaabramov** [comment]
I am not sure how to properly write a test for this case. Any suggestions?

File: core/src/main/scala/fr/davit/akka/http/metrics/core/MeterStage.scala [hunk 1]
```diff
@@ -60,7 +67,10 @@ private[metrics] class MeterStage(metricsHandler: HttpMetricsHandler)
 
       override def onPush(): Unit = {
         val request = grab(requestIn)
-        val id      = request.getAttribute(HttpMetrics.TraceId).get
+        val id = request.attribute(HttpMetrics.TraceId) match {
+          case Some(id) => id
+          case _        => throw MissingTraceIdException
```
Comment (RustedBones) [inline core/src/main/scala/fr/davit/akka/http/metrics/core/MeterStage.scala:72]: When possible we should avoid throwing. In this case we can use the `failStage` method
```suggestion
          case Some(id) => 
            pending += id -> request
            push(requestOut, metricsHandler.onRequest(request))
          case _ => 
            failStage(MissingTraceIdException)
```

**Review by RustedBones — COMMENTED** [review]
For testing you can do smth like
```scala
  it should "throw a MissingTraceIdException if trace-id attribute is missing" in new Fixture {
    requestIn.sendNext(HttpRequest())
    requestIn.expectCancellationWithCause(MissingTraceIdException)
    requestOut.expectError(MissingTraceIdException)
    responseIn.expectCancellationWithCause(MissingTraceIdException)
    responseOut.expectError(MissingTraceIdException)
  }
```

File: core/src/main/scala/fr/davit/akka/http/metrics/core/MeterStage.scala (hunk 1 above)
Comment (aaabramov) [reply core/src/main/scala/fr/davit/akka/http/metrics/core/MeterStage.scala:72]: I am all hands for not throwing exception. Unfortunately, I am not so strong in Akka streams. Will change it

**Comment by aaabramov** [comment]
Added test & replaced exception throwing with `failStage`

**Review by RustedBones — APPROVED** [review]
Thanks a lot @aaabramov !

**Pull request merged**